lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -18,13 +19,38 @@
 #include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 #include "llvm/Support/Debug.h"
 
+#include <iostream>
 using namespace mlir;
 
 #define GEN_PASS_CLASSES
 #include "TPP/Passes.h.inc"
 
 #define DEBUG_TYPE "mlir-map-to-brgemm"
 
+// Look for [p ... p] brgemm [r p p r r]
+static LogicalResult checkVNNIStructure(linalg::LinalgOp linalgOp) {
+  SmallVector<utils::IteratorType> iteratorTypes =
+      linalgOp.getIteratorTypesArray();
+  if (iteratorTypes.size() < 5)
+    return failure();
+  size_t size = iteratorTypes.size() - 1;
+  bool match = linalg::isReductionIterator(iteratorTypes[size]) &&
+               linalg::isReductionIterator(iteratorTypes[size - 1]) &&
+               linalg::isParallelIterator(iteratorTypes[size - 2]) &&
+               linalg::isParallelIterator(iteratorTypes[size - 3]) &&
+               linalg::isReductionIterator(iteratorTypes[size - 4]);
+  if (!match)
+    return failure();
+  size = size - /*BRGEMM loops*/ 4;
```

Comment:
BRGEMM has 3 loops in linalg can we maybe specify VNNI BRGEMM or something similar?